reinforcement_learning/rl_game_server_autopilot/ap-server/app.py

```diff
@@ -0,0 +1,228 @@
+from chalice import Chalice
+import json
+import numpy as np
+from time import gmtime,strftime
+import time
+from datetime import datetime
+from urllib.parse import urlparse, parse_qs
+import boto3
+import sys
+import os
+import requests
+
+#Global Varibale from model
+content_type = "application/json"
+accept = "Accept"
+min_servers=10
+max_servers=100
+new_alloc=0
+region='us-west-2'
+
+sagemaker_region = 'us-west-2'
+sagemaker_client = boto3.client('sagemaker-runtime',region_name=sagemaker_region)
+cloudwatch_region = 'us-west-2'
+cloudwatch_cli = boto3.client('cloudwatch',region_name=cloudwatch_region)
+
+app = Chalice(app_name='autopilot-server')
+
+#TBD Export to environment variables
+endpoint_name ="sagemaker-tensorflow-serving-2019-09-23-20-53-20-237"
+namespace = "autopilot"
+gs_inventory_url = "https://3vwufqstq5.execute-api.us-west-2.amazonaws.com/api/currsine24h"
+
+
+supported_calls={"SupportedCalls":
+                  {
+                     "/predict":"get game-server predictions",
+                     "/currtime":"get the current time in the cloud",
+                     "/currsine1h":"get sine value of 1-hour full positive sine cycle",
+                     "/currsine24h":"get sine value of 24-hour full positive sine cycle",
+                  }
+                }
+
+@app.route('/')
+def index():
+    return supported_calls
+
+def deserializer(item,observations):
+    print('in deserializer '+str(item))
+    item=item.replace("[","")
+    item=item.replace("]","")
+    print('item='+str(item))
+    for num in item.split(" "):
+       if (num):
+         print('num='+str(num))
+         observations.append(float(num))
+
+def get_last_obs_arr(table):
+    print('in get_last_obs_arr {}'.format(table))
+    observations=[]
+    try:
+      response = table.get_item(
+      Key={
+        'key': 'observation'
+      }
+      )
+      x=json.dumps(response['Item'])
+      item = response['Item']['value']
+      deserializer(item,observations)
+      print('observations='+str(observations))
+      return observations
+    except Exception as e:
+      print('Error {}'.format(e))
```

Comment:
Should we be  consuming this exception? In case of Throttling/Missing Key exception, this function would return None, and behaviour would be unexpected. (I see a json.dumps being called on the return of this function, and it swallows None object returning 'null' string, with any exception)